ste/sender-blockBlob.go

```diff
@@ -153,19 +153,38 @@ func (s *blockBlobSenderBase) Epilogue() {
 	// https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers
 	if jptm.IsLive() && s.destBlobTier != azblob.AccessTierNone {
 		// Set the latest service version from sdk as service version in the context.
-		ctxWithLatestServiceVersion := context.WithValue(jptm.Context(), ServiceAPIVersionOverride, azblob.ServiceVersion)
-		_, err := s.destBlockBlobURL.SetTier(ctxWithLatestServiceVersion, s.destBlobTier, azblob.LeaseAccessConditions{})
-		if err != nil {
-			if s.jptm.Info().S2SSrcBlobTier != azblob.AccessTierNone {
-				s.jptm.LogTransferInfo(pipeline.LogError, s.jptm.Info().Source, s.jptm.Info().Destination, "Failed to replicate blob tier at destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-				s2sAccessTierFailureLogStdout.Do(func() {
-					glcm := common.GetLifecycleMgr()
-					glcm.Error("One or more blobs have failed blob tier replication at the destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-				})
+		ctxWithLatestServiceVersion := context.WithValue(s.jptm.Context(), ServiceAPIVersionOverride, azblob.ServiceVersion)
```

Comment:
This code is now duplicated in the blockBLob and page blob senders. At about 35 lines, that's more duplication than we'd normally be comfortable with.  Can you refactor it into a routine that exists in one place, and gets called by the block and page blob senders?

Reply:
PS. The reason is not just that these are 35 duplicated lines.  The other part of the reason is that we have to be very vigilant about duplicated code. If we're not careful about it, we'll end up with a whole lot of duplication and that leads to maintenance problems. (As we saw, prior to the various refactorings of 2019, since before those refactorings duplication was a problem).

Reply:
The code is now de-duped.